extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -0,0 +1,191 @@
+'use strict'; // eslint-disable-line
+
+const Logger = require('werelogs').Logger;
+const errors = require('arsenal').errors;
+const MongoClient = require('arsenal').storage
+    .metadata.mongoclient.MongoClientInterface;
+const { usersBucket } = require('arsenal').constants;
+const BackbeatConsumer = require('../../lib/BackbeatConsumer');
+const QueueEntry = require('../../lib/models/QueueEntry');
+const DeleteOpQueueEntry = require('../../lib/models/DeleteOpQueueEntry');
+const BucketQueueEntry = require('../../lib/models/BucketQueueEntry');
+const BucketMdQueueEntry = require('../../lib/models/BucketMdQueueEntry');
+const ObjectQueueEntry = require('../../lib/models/ObjectQueueEntry');
+
+// TODO - ADD PREFIX BASED ON SOURCE
+// april 6, 2018
+
+/**
+ * @class MongoQueueProcessor
+ *
+ * @classdesc Background task that processes entries from the
+ * ingestion for kafka queue and pushes entries to mongo
+ */
+class MongoQueueProcessor {
+
+    /**
+     * @constructor
+     * @param {Object} zkConfig - zookeeper configuration object
+     * @param {string} zkConfig.connectionString - zookeeper connection string
+     *   as "host:port[/chroot]"
+     * @param {Object} mongoProcessorConfig - mongo processor configuration
+     *   object
+     * @param {String} mongoProcessorConfig.topic - topic name
+     * @param {String} mongoProcessorConfig.groupId - kafka
+     *   consumer group ID
+     * @param {Integer} mongoProcessorConfig.retryTimeoutS -
+     *   number of seconds before giving up retries of an entry status
+     *   update
+     * @param {Object} mongoClientConfig - config for connecting to mongo
+     */
+    constructor(zkConfig, mongoProcessorConfig, mongoClientConfig) {
+        this.zkConfig = zkConfig;
+        this.mongoProcessorConfig = mongoProcessorConfig;
+        this._consumer = null;
+        this.logger =
+            new Logger('Backbeat:MongoProcessor');
+        this.mongoClientConfig = mongoClientConfig;
+        this.mongoClientConfig.logger = this.logger;
+        this._mongoClient = new MongoClient(this.mongoClientConfig);
+    }
+
+    /**
+     * Start kafka consumer
+     *
+     * @return {undefined}
+     */
+    start() {
+        this.logger.info('starting mongo queue processor');
+        this._mongoClient.setup(err => {
+            if (err) {
+                this.logger.error('could not connect to MongoDB', { err });
+            }
+        });
+        this._consumer = new BackbeatConsumer({
+            // zookeeper: { connectionString: this.zkConfig.connectionString },
+            topic: this.mongoProcessorConfig.topic,
+            groupId: `${this.mongoProcessorConfig.groupId}`,
+            // Must always have concurrency of 1 so writes are in order
+            kafka: { hosts: '127.0.0.1:9092' },
+            concurrency: 1,
```

Comment:
I'm rather confident we can get rid of this limit for ingestion (likely cut&paste left-over, will remove w/ @rahulreddy 's benediction)